lib/front_matter_parser/writer.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require_relative "parser"
+require_relative "dumper"
+require_relative "syntax_writer"
+
+module FrontMatterParser
+  # Entry point to parse a front matter from a string or file.
+  class Writer < Parser
```

Comment:
I don't like making `Writer` a descendant of `Parser`. They are more like siblings.. I think you make it just in order to reuse the `syntax_from_pathname` method. If so, we could move it to `FrontMatter` or a custom helper.